### PR TITLE
Separate kpsr codegen

### PR DESCRIPTION
**Pull request by mharshe**

### CODE REVIEW

File: cpp_testing/modules/CMakeLists.txt
Comment (mharshe) [inline cpp_testing/modules/CMakeLists.txt:1]: This file will be deleted

File: cpp_testing/modules/codegen/CMakeLists.txt
Comment (mharshe) [inline cpp_testing/modules/codegen/CMakeLists.txt:1]: This file will be deleted too. The components of this have been moved to CMakeLists.txt in the tests folders

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    kpsr_codegen = kpsr_codegen.__main__:main
+
+[options.packages.find]
+exclude = tests*
+
+[options.package_data]
+kpsr_codegen =
+        conf/*
+        templates/*
+        cmake/*
```
Comment (mharshe) [inline setup.cfg:32]: This adds the cmake/KpsrCodeGen.cmake to the package. But maybe that file should be in kpsr-core ? I'm not sure how to have this .cmake file installed in the correct location for cpp projects/cmake to find.

**Review by mharshe — COMMENTED** [review]
Most changes are to the setup.cfg file and the CMakeLists.txt files for testing the codegeneration tool

File: setup.cfg (hunk 1 above)
Comment (pablogh-2000) [reply setup.cfg:32]: We can create a post install command in the setup.py:
https://stackoverflow.com/questions/20288711/post-install-script-with-python-setuptools
I'll add a JIRA for this.

**Review by pablogh-2000 — APPROVED** [review]

**Pull request merged**